include/wabt/type.h

```diff
@@ -65,7 +65,28 @@ class Type {
   Type(Enum e, Index type_index) : enum_(e), type_index_(type_index) {
     assert(e == Enum::Reference);
   }
-  constexpr operator Enum() const { return enum_; }
+  constexpr Enum code() const { return enum_; }
+
+  bool operator==(const Type& other) const {
+    return enum_ == other.enum_ && type_index_ == other.type_index_;
+  }
+
+  bool operator!=(const Type& other) const {
+    return enum_ != other.enum_ || type_index_ != other.type_index_;
+  }
+
+  bool operator<(const Type& other) const {
+    return (enum_ != other.enum_) ? enum_ < other.enum_
+                                  : type_index_ < other.type_index_;
+  }
+
+  bool operator==(const Enum& other) const {
+    return enum_ == other;
+  }
+
+  bool operator!=(const Enum& other) const {
+    return enum_ != other;
+  }
```

Comment:
Do we actually need to these operators for comparing to Enum?

Reply:
It looks like this is not needed.

Reply:
I still see these comparisons with Enum.

Reply:
I only removed the !=. I thought the compiler can generate a better code when comparing to an enum. Anyway, it looks like the source code compiles without them.